frontend/test/resolution/testScopeResolve.cpp

```diff
@@ -570,6 +570,134 @@ static void test13() {
   assert(guard.realizeErrors());
 }
 
+// There's specal handling for the rightmost field access. Make sure this
+// special handling properly handles super.
+static void test14() {
+  printf("test14\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::get(context, "input.chpl");
+  std::string contents = R""""(
+      module A {
+        var x : int;
+        module B {
+          import this.super as C;
+          var y = C.x;
+        }
+      }
+   )"""";
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parseToplevel(context, path);
+
+  const Variable* x = findVariable(vec, "x");
+  assert(x);
+  const Variable* y = findVariable(vec, "y");
+  assert(y);
+
+  const ResolvedExpression& reY = scopeResolveIt(context, y->initExpression());
+  assert(reY.toId() == x->id());
+}
+
+
+// Make sure that the dot-expression handling of "this" works in addition
+// to the idenifier-expression handling of "this". Technically this is
+// redundant, but our goal is to issue a warning, not fail to resolve in
```

Comment:
This says our goal is to issue a warning, but I don't see a check for a warning in the test?

Reply:
The warning isn't there yet, it's one of the still-failing scope resolution issues and requires more infrastructure.